File: Python/35.Search Insert Position.py
#求出给定目标数字在有序数组中应该存放的位置
class Solution(object):
    def searchInsert(self, nums, target):
        l , r = 0 , len(nums)
        while l<r:
            mid = (l+r)//2
            if nums[mid]<target:
                l = mid+1
            else:
                r = mid
        return l



s = Solution()
print(s.searchInsert([5,7,7,8,8,10],8))

File: C++/Binary Search/35.Search Insert Position.cpp
/*
向排序数组中插入一个数，找出应该插入的位置。
*/
/*
思路：二分注意循环退出条件。l==r时再进一次循环，那么结束时l的位置就是应该插入的位置。
*/
#include <iostream>
#include <cstring>
#include <vector>
using namespace std;

class Solution {
public:
	int searchInsert(vector<int>& nums, int target) {
        int l = 0, r = nums.size()-1;
        while(l <= r)
        {
        	int mid = (l + r) / 2;
        	if(nums[mid] < target)	l = mid+1;
        	else	r = mid-1;
        }
        return l;
    }
};

int main()
{
	Solution s;
	vector<int> nums{ 1,3,5,6 };
	cout<<s.searchInsert(nums, 2)<<endl;
}

